homeassistant/components/lock/__init__.py

```diff
@@ -87,40 +87,36 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     return True
 
 
-async def _async_lock(entity: LockEntity, service_call: ServiceCall) -> None:
-    """Lock the lock."""
-    code: str = service_call.data.get(
-        ATTR_CODE, entity._lock_option_default_code  # pylint: disable=protected-access
-    )
+@callback
+def _add_default_code(entity: LockEntity, service_call: ServiceCall) -> dict[Any, Any]:
+    code: str = service_call.data.get(ATTR_CODE, "")
+    if not code:
+        code = entity._lock_option_default_code  # pylint: disable=protected-access
     if entity.code_format_cmp and not entity.code_format_cmp.match(code):
         raise ValueError(
             f"Code '{code}' for locking {entity.entity_id} doesn't match pattern {entity.code_format}"
         )
-    await entity.async_lock(**remove_entity_service_fields(service_call))
+    data = remove_entity_service_fields(service_call)
+    if code:
+        data[ATTR_CODE] = code
+    elif ATTR_CODE in data:
+        data.pop(ATTR_CODE)
```

Comment:
```suggestion
    data.pop(ATTR_CODE)
    if code:
        data[ATTR_CODE] = code
```

Reply:
I moved the `pop` to where we get the code in the first place.